helix-term/src/commands/dap.rs

```diff
@@ -475,19 +475,32 @@ pub fn dap_variables(cx: &mut Context) {
 
     if debugger.thread_id.is_none() {
         cx.editor
-            .set_status("Cannot access variables while target is running");
+            .set_status("Cannot access variables while target is running.");
         return;
     }
     let (frame, thread_id) = match (debugger.active_frame, debugger.thread_id) {
         (Some(frame), Some(thread_id)) => (frame, thread_id),
         _ => {
             cx.editor
-                .set_status("Cannot find current stack frame to access variables");
+                .set_status("Cannot find current stack frame to access variables.");
             return;
         }
     };
 
-    let frame_id = debugger.stack_frames[&thread_id][frame].id;
+    let thread_frame = debugger.stack_frames.get(&thread_id);
+    if thread_frame.is_none() {
+        cx.editor
+            .set_error("Failed to get stack frame for thread: {thread_id}");
+        return;
+    }
+    let stack_frame = thread_frame.unwrap().get(frame);
+    if stack_frame.is_none() {
+        cx.editor
+            .set_error("Failed to get stack frame for thread {thread_id} and frame {frame}.");
+        return;
+    }
```

Comment:
Avoid unwraps, follow the same pattern as above:

```rust
let thread_frame = debugger.stack_frames.get(&thread_id) {
  Some(thread_frame) => thread_frame,
  None => {
     cx.editor.set_error(...)    
     return;
   }
};
```

Reply:
And this actually returns, breaking further execution, right? That's actually really cool, didn't know you could write it as such! Thanks a lot, will implement promptly!